sql: make the interface of SHOW and EXPLAIN more consistent

knz opened the pull request:
Fixes #26840.

Summary: this PR mainly updates the column labels produced for the
output of several SHOW and EXPLAIN statements. The data produced is
generally left unchanged, with two exceptions detailed below.

## Motivation

SHOW and EXPLAIN statements in CockroachDB provide convenient views
over database metadata and query execution.

They constitute a public-facing API and are prominently featured in
documentation and tutorials.

It is thus important that their interface is polished and
consistent. This ensures users can more effectively construct a
reliable mental model of how to use CockroachDB: more regularity and
consistency make the software easier to use. Also, consistency and
more generally design polish give a good impression to users and
increases the chance of reputation-based popularity gains.

In addition, since mid-2017 CockroachDB enables queries to apply SQL
to refine and modify the output of SHOW or EXPLAIN statements. This is
used extensively in tests and during troubleshooting, and incidentally
when teaching CockroachDB. It is thus important that the "in-SQL
interface" of these statements makes the construction of surrounding
queries convenient too.

## Background

As of june 2018 CockroachDb currently supports 39 SHOW and EXPLAIN
statements. They have been added incrementally and organically as need
arised. The following table reports the various syntax forms sorted by
creation order, with their column interface prior to this PR.

| Syntax form                | date added | old parameter list                                                                                                                                                              |
|----------------------------|------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| SHOW COLUMNS               | 2015-06    | "Field", "Type", "Null", "Default"                                                                                                                                              |
| SHOW DATABASES             | 2015-06    | "Database"                                                                                                                                                                      |
| SHOW INDEX                 | 2015-06    | "Table", "Name", "Unique", "Seq", "Column", "Direction", "Storing", "Implicit"                                                                                                  |
| SHOW TABLES                | 2015-06    | "Table"                                                                                                                                                                         |
| EXPLAIN (PLAN)             | 2015-08    | "Tree", "Field", "Description"                                                                                                                                                  |
| SHOW GRANTS                | 2015-08    | "Database", "Schema", "User", "Privileges"                                                                                                                                      |
| SHOW SESSION (var)         | 2015-08    | (var)                                                                                                                                                                           |
| SHOW TRACE                 | 2016-02    | timestamp, age, message, tag, loc, operation, span                                                                                                                              |
| SHOW CREATE TABLE          | 2016-04    | "Table", "ShowCreate"                                                                                                                                                           |
| SHOW CONSTRAINTS           | 2016-07    | "Table", "Name", "Type", "Column(s)", "Details"                                                                                                                                 |
| SHOW CREATE VIEW           | 2016-10    | "View", "ShowCreate"                                                                                                                                                            |
| SHOW SESSION ALL           | 2016-10    | variable, value (*)                                                                                                                                                             |
| SHOW TRANSACTION STATUS    | 2016-11    | transaction_status                                                                                                                                                              |
| EXPLAIN (DISTSQL)          | 2017-02    | "Automatic", "JSON", "URL"                                                                                                                                                      |
| SHOW RANGES                | 2017-03    | "Start Key", "End Key", "Range ID", "Replicas", "Lease Holder"                                                                                                                  |
| SHOW ALL CLUSTER SETTINGS  | 2017-04    | name, current_value, type, description (*)                                                                                                                                      |
| SHOW CLUSTER SETTING (var) | 2017-04    | (var)                                                                                                                                                                           |
| SHOW FINGERPRINTS          | 2017-05    | "index", fingerprint                                                                                                                                                            |
| SHOW BACKUP (default)      | 2017-06    | database, "table", start_time, end_time, size_bytes                                                                                                                             |
| SELECT FROM [...]          | 2017-06    |                                                                                                                                                                                 |
| SHOW JOBS                  | 2017-06    | id, type, description, username, status, created, started, finished, modified, fraction_completed, error, coordinator_id                                                        |
| SHOW QUERIES               | 2017-06    | query_id, node_id, username, start, query, client_address, application_name, distributed, phase (*)                                                                             |
| SHOW SESSIONS              | 2017-06    | node_id, session_id, username, client_address, application_name, active_queries, last_active_query, session_start, oldest_query_start, kv_txn, alloc_bytes, max_alloc_bytes (*) |
| SHOW ZONE CONFIGURATION    | 2017-10    | id, cli_specifier, config_yaml, config_proto                                                                                                                                    |
| SHOW COMPACT TRACE         | 2017-11    | age, message, tag, operation                                                                                                                                                    |
| SHOW USERS                 | 2017-11    | username                                                                                                                                                                        |
| SHOW HISTOGRAM             | 2017-12    | upper_bound, num_range, num_eq                                                                                                                                                  |
| SHOW ROLES                 | 2017-12    | rolename                                                                                                                                                                        |
| SHOW STATISTICS            | 2017-12    | name, columns, created_at, row_count, distinct_count, null_count, histogram_id                                                                                                  |
| SHOW SYNTAX                | 2017-12    | field, text                                                                                                                                                                     |
| SHOW CREATE SEQUENCE       | 2018-01    | "Sequence", "ShowCreate"                                                                                                                                                        |
| SHOW GRANTS ON ROLE        | 2018-01    | role, member, "isAdmin"                                                                                                                                                         |
| SHOW REPLICA TRACE         | 2018-01    | timestamp, node_id, store_id, replica_id                                                                                                                                        |
| SHOW SCHEMAS               | 2018-02    | "Schema"                                                                                                                                                                        |
| SHOW STATISTICS USING JSON | 2018-04    | statistics                                                                                                                                                                      |
| EXPLAIN ANALYZE            | 2018-05    | "Automatic", "JSON", "URL"                                                                                                                                                      |
| EXPLAIN (OPT)              | 2018-06    | text                                                                                                                                                                            |
| SHOW BACKUP (files)        | 2018-06    | path, start_pretty, end_pretty, start_key, end_key, size_bytes, rows                                                                                                            |
| SHOW BACKUP (ranges)       | 2018-06    | start_pretty, end_pretty, start_key, end_key                                                                                                                                    |

The following notes apply:

- column labels are double quoted when they are not valid identifiers,
  either because of casing or because they are restricted SQL keywords.
- `(var)` for SHOW CLUSTER SETTING and SHOW SESSION indicates the column
  of the result is named after the variable being shown.
- `(*)` in the column list for SHOW QUERIES, SHOW SESSIONS, SHOW
  SESSION ALL and SHOW ALL CLUSTER SETTINGS is inherited directly from
  a `crdb_internal` table. This is discussed below.

## Problem statement

The table above reveals the 4 general problems addressed in this PR:

- the result column labels are inconsistently cased. The inconsistency
  itself is hurtful to UX and may give a poor impression of
  CockroachDB's overall design.

- the same conceptual data is named using different words in different
  statements. For example SHOW GRANTS uses "User" but SHOW USERS uses
  "username". SHOW TABLES uses "Table" but "SHOW STATISTICS" uses
  "name". This makes it harder for users to construct a reusable,
  simple conceptual model of how to inspect CockroachDB.

- the use of capitals or restricted keywords in column labels mandate
  double quotes when using the column in a surrounding SQL query with
  `SELECT FROM [...]`. It also creates doubt and confusion as a user
  may remember the label of a column but not which casing to use to
  refer to it.

- a few statement are simple pass-through views to an underlying
  `crdb_internal` table without fixing the list of result columns. If
  an engineer were to modify the `crdb_internal` table, they would
  thus modify the public-facing, documented SQL API of CockroachDB
  without realizing.

## Detailed changes

- SHOW COLUMNS:
  - `Field` -> `column_name` (casing, disambiguation, consistency with `information_schema`)
  - `Type` -> `data_type` (casing, disambiguation, consistency with `information_schema`)
  - `Null` -> `is_nullable` (casing, consistency with `information_schema`)
  - `Default` -> `generation_expression` (casing, keyword avoidance, consistency with `information_schema`)

- SHOW DATABASES:
  - `Database` -> `database_name` (casing, disambiguation)

- SHOW INDEX:
  - `Table` -> `table_name` (casing, keyword avoidance, disambiguation)
  - `Index` -> `index_name` (casing, keyword avoidance, disambiguation)
  - `Unique` -> `non_unique` (casing, keyword avoidance, consistency with `information_schema`)
    Note: this inverts the boolean meaning.
  - `Seq` -> `seq_in_index` (casing, consistency with `information_schema`)
  - `Column` -> `column_name` (casing, disambiguation, consistency with `information_schema`)
  - `Direction` -> `direction` (casing)
  - `Storing` -> `storing` (casing)
  - `Implicit` -> `implicit` (casing)

- `SHOW TABLES`
  - `Table` -> `table_name` (casing, keyword avoidance, disambiguation)

- `EXPLAIN (PLAN)`
  - `Tree` -> `tree` (casing)
  - `Field` -> `field` (casing)
  - `Description` -> `description` (casing)

- `SHOW GRANTS`
  - `Database` -> `database_name` (casing, consistency, disambiguation)
  - `Schema` -> `schema_name` (casing, consistency)
  - `User` -> `grantee` (casing, consistency with `information_schema`, also because it includes roles not just users)
  - `Privileges` -> `privilege_type` (casing, consistency with `information_schema`)

- `SHOW TRACE`
  - `loc` -> `location` (avoidance of abbreviation)

- `SHOW CREATE TABLE`, `SHOW CREATE VIEW`, `SHOW CREATE SEQUENCE`
  - make the TABLE/VIEW/SEQUENCE keyword optional: they all run the same code and
    the users should not have to remember whether their object is a table
    or a view to run SHOW CREATE.
  - `Table`/`View`/`Sequence` -> `table_name` (casing, consistency, keyword avoidance, disambiguation)
  - `ShowCreate` -> `create_statement` (casing, consistency)

- `SHOW CONSTRAINTS`
  - `Table` -> `table_name` (casing, consistency, keyword avoidance, disambiguation)
  - `Name` -> `constraint_name` (casing, disambiguation)
  - `Type` -> `constraint_type` (casing, disambiguation)
  - `Column(s)`/`Details` -> `details`, `validated` (casing, simplification)

- `SHOW SESSION ALL`
  - fixed the set of output columns instead of passing `crdb_internal` through.

- `EXPLAIN (DISTSQL)`, `EXPLAIN ANALYZE`
  - `Automatic` -> `automatic` (casing)
  - `URL` -> `url` (casing)
  - `JSON` -> `json` (casing)

- `SHOW RANGES`
  - `Start Key` -> `start_key` (casing, word separation)
  - `End Key` -> `end_key`  (casing, word separation)
  - `Range ID` -> `range_id`  (casing, word separation)
  - `Replicas` -> `replicas`  (casing)
  - `Lease Holder` -> `lease_holder`  (casing, word separation)

- `SHOW ALL CLUSTER SETTINGS`
  - `name` -> `variable` (consistency with SHOW SESSION ALL)
  - `current_value` -> `value` (consistency with SHOW SESSION ALL)
  - `type` -> `setting_type` (disambiguation)

  This also renames the columns in `crdb_internal.cluster_settings`
  accordingly.

- `SHOW FINGERPRINTS`
  - `index` -> `index_name` (keyword avoidance, disambiguation)

- `SHOW BACKUP`
  - `database` -> `database_name` (disambiguation)
  - `table` -> `table_name` (keyword avoidance, disambiguation)

- `SHOW JOBS`
  - `id` -> `job_id` (disambiguation)
  - `type` -> `job_type` (disambiguation)
  - `username` -> `user_name` (space between words)

  This also renames the columns in `crdb_internal.jobs`
  accordingly.

- `SHOW QUERIES`
  - `username` -> `user_name` (space between words)
  - fixed the set of output columns instead of passing `crdb_internal` through.

- `SHOW SESSIONS`
  - `username` -> `user_name` (space between words)
  - fixed the set of output columns instead of passing `crdb_internal` through.

- `SHOW ZONE CONFIGURATION`
  - `id` -> `zone_id` (disambiguation)
  - `config_proto` -> `config_protobuf` (abbreviation avoidance)

- `SHOW HISTOGRAM`
  - `num_range` -> `range_rows` (abbreviation avoidance)
  - `num_eq` -> `equal_rows` (abbreviation avoidance)

- `SHOW ROLES`
  - `rolename` -> `role_name` (space between words, disambiguation)

- `SHOW STATISTICS`
  - `name` -> `table_name` (disambiguation)
  - `columns` -> `column_names` (disambiguation)
  - `created_at` -> `created` (consistency with SHOW JOBS)

- `SHOW GRANTS ON ROLE`
  - `role` -> `role_name` (consistency with SHOW ROLES)
  - `isAdmin` -> `is_admin` (casing, word separator)

- `SHOW SCHEMAS`
  - `Schema` -> `schema_name` (casing, consistency, disambiguation)

- `SHOW USERS`
  - `username` -> `user_name` (space between words)

- `EXPLAIN ANALYZE`
  - `Automatic` column removed (not relevant for this syntax form)
  - `JSON` -> `json` (casing)
  - `URL` -> `url` (casing)

## Methodology

The following principles have guided the work in this PR.

- the casing of column labels must be consistent.

- the same concept in different statements should be named using the
  same word.

  For example, `name` -> `variable` in SHOW ALL CLUSTER SETTINGS for
  consistency with SHOW SESSION ALL.

- the labels must be usable by a surrounding query without mandating
  quotes.

  For example, `"Start Key"` -> `start_key`

- underscores to separate words for consistency with
  `information_schema`.

  For example, `"Start Key"` -> `start_key` instead of `startkey`

- avoid abbreviations unless most uses of the word in spoken English
  use the abbreviation already.

  For example, `id` for "identifier" is OK, but `loc` for "location"
  is not.

- for columns akin to "primary keys" (that identify the object being
  described by the remaining columns), words whose meaning is highly
  overloaded must be disambiguated in the label.

  For example, `id` -> `zone_id` or `job_id`; `name` -> `column_name`.

- for database objects that can be referred to by two or more types of
  handles (e.g. by-ID or by-name), the column label must disambiguate
  which type of handle is being reported. This makes it possible to
  later report the other handles in additional columns.

  For example, `table` -> `table_name`, so that `table_id` can be
  added later.

- labels that reproduce data already present in an
  `information_schema` table should use the same labels as the
  `information_schema` table provided it matches the principles above
  already.

A later PR will codify these principles by additions in the `docs/`
directory and possibly a linter.

## Summary of statement interfaces after the changes

| Syntax form                | new parameter list                                                                                                                                                               |
|----------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| EXPLAIN (DISTSQL)          | **automatic, json, url**                                                                                                                                                         |
| EXPLAIN (OPT)              | text                                                                                                                                                                             |
| EXPLAIN (PLAN)             | **tree, field, description**                                                                                                                                                     |
| EXPLAIN ANALYZE            | **json, url**                                                                                                                                                                    |
| SHOW ALL CLUSTER SETTINGS  | **variable, value, setting_type**, description                                                                                                                                   |
| SHOW BACKUP (default)      | **database_name**, **table_name**, start_time, end_time, size_bytes                                                                                                              |
| SHOW BACKUP (files)        | path, start_pretty, end_pretty, start_key, end_key, size_bytes, rows                                                                                                             |
| SHOW BACKUP (ranges)       | start_pretty, end_pretty, start_key, end_key                                                                                                                                     |
| SHOW CLUSTER SETTING (var) | (var)                                                                                                                                                                            |
| SHOW COLUMNS               | **column_name, data_type, is_nullable, generation_expression**                                                                                                                   |
| SHOW COMPACT TRACE         | age, message, tag, operation                                                                                                                                                     |
| SHOW CONSTRAINTS           | **table_name, constraint_name, constraint_type, details, validated**                                                                                                             |
| SHOW CREATE SEQUENCE       | **table_name, create_statement**                                                                                                                                                 |
| SHOW CREATE TABLE          | **table_name, create_statement**                                                                                                                                                 |
| SHOW CREATE VIEW           | **table_name, create_statement**                                                                                                                                                 |
| SHOW DATABASES             | **database_name**                                                                                                                                                                |
| SHOW FINGERPRINTS          | **index_name**, fingerprint                                                                                                                                                      |
| SHOW GRANTS                | **database_name, schema_name, grantee, privilege_type**                                                                                                                          |
| SHOW GRANTS ON ROLE        | **role_name**, member, **is_admin**                                                                                                                                              |
| SHOW HISTOGRAM             | upper_bound, **range_rows**, **equal_rows**                                                                                                                                      |
| SHOW INDEX                 | **table_name, index_name, non_unique, seq_in_index, column_name, direction, storing, implicit**                                                                                  |
| SHOW JOBS                  | **job_id**, **job_type**, description, **user_name**, status, created, started, finished, modified, fraction_completed, error, coordinator_id                                    |
| SHOW QUERIES               | query_id, node_id, **user_name**, start, query, client_address, application_name, distributed, phase                                                                             |
| SHOW RANGES                | **start_key, end_key, range_id, replicas, lease_holder**                                                                                                                         |
| SHOW REPLICA TRACE         | timestamp, node_id, store_id, replica_id                                                                                                                                         |
| SHOW ROLES                 | **role_name**                                                                                                                                                                    |
| SHOW SCHEMAS               | **schema_name**                                                                                                                                                                  |
| SHOW SESSION (var)         | (var)                                                                                                                                                                            |
| SHOW SESSION ALL           | variable, value                                                                                                                                                                  |
| SHOW SESSIONS              | node_id, session_id, **user_name**, client_address, application_name, active_queries, last_active_query, session_start, oldest_query_start, kv_txn, alloc_bytes, max_alloc_bytes |
| SHOW STATISTICS            | **table_name**, **column_names**, **created**, row_count, distinct_count, null_count, histogram_id                                                                               |
| SHOW STATISTICS USING JSON | statistics                                                                                                                                                                       |
| SHOW SYNTAX                | field, text                                                                                                                                                                      |
| SHOW TABLES                | **table_name**                                                                                                                                                                   |
| SHOW TRACE                 | timestamp, age, message, tag, **location**, operation, span                                                                                                                      |
| SHOW TRANSACTION STATUS    | transaction_status                                                                                                                                                               |
| SHOW USERS                 | **user_name**                                                                                                                                                                    |
| SHOW ZONE CONFIGURATION    | **zone_id**, cli_specifier, config_yaml, **config_protobuf**                                                                                                                     |

## Backward compatibility

This technically constitutes a backward-incompatible change (change in
public API which may theoretically confuse existing client code). The
reason why the change is nonetheless worth adopting is fourfold:

- automated tools, especially those designed for compatibility with a
  pg server, are more likely to use `pg_catalog` and
  `information_schema` tables instead.

- the EXPLAIN and SHOW statements are still advertised and
  designed/implemented as "porcelain" [1] meant for use by humans.

- There has not been any commitment to their interface for
  backward-compatibility (other than a general guideline that
  evolutions do not remove columns already known to be useful). In
  fact, over time, several SHOW statements have evolved their
  interface and this has not yet been picked up during reviews and QA
  as a source of backward compatibility concern.

- The general UX improvement and its potential benefits for ease of
  adoption outweighs backward compatibility concerns.

[1]: https://stackoverflow.com/questions/6976473/what-does-the-term-porcelain-mean-in-git#6978402

knz committed:
sql: rationalize SHOW COLUMNS

The output columns of SHOW COLUMNS are renamed as follows:

- `Field` -> `column_name` (casing, disambiguation, consistency with `information_schema`)
- `Type` -> `data_type` (casing, disambiguation, consistency with `information_schema`)
- `Null` -> `is_nullable` (casing, consistency with `information_schema`)
- `Default` -> `column_default` (casing, keyword avoidance,
- consistency with `information_schema`)

New output column: `generation_expression` for computed columns in the
table.

Release note (sql change): The labels of the output columns of `SHOW
COLUMNS` are renamed for consistency with `information_schema`. A new
column `generation_expression` reports the expression used for
computed columns.

knz committed:
sql: rationalize SHOW CREATE

SHOW CREATE TABLE/VIEW/SEQUENCE really run the same code. Also, the
users should not have to remember whether the object they want to
inspect is a table or a view to run SHOW CREATE. So this patch makes
the final keyword optional, such that `SHOW CREATE foo` is now
possible.

The output labels are also renamed as follows:

- `Table`/`View`/`Sequence` -> `table_name` (casing, consistency, keyword avoidance, disambiguation)
- `ShowCreate` -> `create_statement` (casing, consistency)

Release note (sql change): The `SHOW CREATE` statement has been
simplified and can be used equivalently on tables, views and sequences
without having to specify the type of object to inspect.

Release note (sql change): The output labels of `SHOW CREATE` have
been renamed for consistency with other SHOW statements.

knz committed:
sql: rationalize SHOW DATABASES

- `Database` -> `database_name` (casing, disambiguation)

Release note (sql change): The column label of the output of `SHOW
DATABASE` is renamed for consistency with other SHOW statements.

knz committed:
sql: rationalize SHOW TABLES

The column label is renamed as follows:

- `Table` -> `table_name` (casing, keyword avoidance, disambiguation)

Release note (sql change): The column label of the output of `SHOW
TABLES` is renamed for consistency with other SHOW statements.

knz committed:
sql: rationalize SHOW GRANTS

The column labels are renamed as follows:

- `Database` -> `database_name` (casing, consistency, disambiguation)
- `Schema` -> `schema_name` (casing, consistency)
- `Table` -> `table_name` (casing, consistency)
- `User` -> `grantee` (casing, consistency with `information_schema`, also because it includes roles not just users)
- `Privileges` -> `privilege_type` (casing, consistency with `information_schema`)

Release note (sql change): The labels of the output columns of `SHOW
GRANTS` have been renamed for consistency with `information_schema`
and other SHOW statements.

knz committed:
sql: rationalize SHOW INDEX

The output columns are renamed as follows:

- `Table` -> `table_name` (casing, keyword avoidance, disambiguation)
- `Index` -> `index_name` (casing, keyword avoidance, disambiguation)
- `Unique` -> `non_unique` (casing, keyword avoidance, consistency with `information_schema`)
  Note: this inverts the boolean meaning.
- `Seq` -> `seq_in_index` (casing, consistency with `information_schema`)
- `Column` -> `column_name` (casing, disambiguation, consistency with `information_schema`)
- `Direction` -> `direction` (casing)
- `Storing` -> `storing` (casing)
- `Implicit` -> `implicit` (casing)

Release note (sql change): The output labels of `SHOW INDEX` (= `SHOW
INDEXES`) are renamed for consistency with other SHOW statements and
`information_schema`.

knz committed:
sql: rationalize SHOW QUERIES

Prior to this patch, SHOW QUERIES would be a pass-through to
`crdb_internal.{cluster,node}_queries`. This is susceptible to API
breakage when the (internal, unstable) vtable is updated.

This patch removes this risk by fixing the columns produced by SHOW
QUERIES.

The output labels are also renamed as follows:

- `username` -> `user_name` (space between words)

Release note (sql change): The output labels of `SHOW QUERIES` have
been renamed for consistency with other SHOW statements.

knz committed:
sql: rationalize SHOW EXPERIMENTAL_RANGES

The output labels have been renamed as follows:

- `Start Key` -> `start_key` (casing, word separation)
- `End Key` -> `end_key`  (casing, word separation)
- `Range ID` -> `range_id`  (casing, word separation)
- `Replicas` -> `replicas`  (casing)
- `Lease Holder` -> `lease_holder`  (casing, word separation)

Release note (sql change): The output labels of `SHOW
EXPERIMENTAL_RANGES` (a.k.a. `SHOW TESTING_RANGES`) have been renamed
for consistency with other SHOW statements.

knz committed:
sql: rationalize SHOW GRANTS ON ROLE

The output labels are renamed as follows:

- `role` -> `role_name` (consistency with SHOW ROLES)
- `isAdmin` -> `is_admin` (casing, word separator)

Release note (sql change): The output labels of `SHOW GRANTS ON ROLE`
are renamed for consistency with other SHOW statements.

knz committed:
sql: rationalize SHOW ROLES

The output label is renamed as follows:

- `rolename` -> `role_name` (space between words, disambiguation)

Release note (sql change): The output labels for `SHOW ROLES` are
renamed for consistency with the other SHOW statements.

knz committed:
sql: rationalize SHOW SCHEMAS

Output label updated as follows

- `Schema` -> `schema_name` (casing, consistency, disambiguation)

Release note (sql change): The output label of SHOW SCHEMAS is updated
for consistency with other SHOW statements.

knz committed:
sql: rationalize SHOW SESSIONS

Prior to this patch, SHOW SESSIONS would be a pass-through to
`crdb_internal.{cluster,node}_sessions`. This is susceptible to API
breakage when the (internal, unstable) vtable is updated.

This patch removes this risk by fixing the columns produced by SHOW
SESSIONS.

The output labels are also renamed as follows:

- `username` -> `user_name` (space between words)

Release note (sql change): The output labels of `SHOW SESSIONS` have
been renamed for consistency with other SHOW statements.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/27098)
<!-- Reviewable:end -->

knz committed:
sql: rationalize EXPLAIN

The output labels are renamed as follows:

- `Tree` -> `tree` (casing)
- `Field` -> `field` (casing)
- `Type` -> `node_type` (casing, disambiguation)
- `Columns` -> `columns` (casing)
- `Ordering` -> `ordering` (casing)
- `Description` -> `description` (casing)
- `Automatic` -> `automatic` (casing)
- `URL` -> `url` (casing)
- `JSON` -> `json` (casing)

Release note (sql change): The labels of the output columns of
`EXPLAIN` have been renamed for consistency with other SHOW
statements.

knz committed:
sql: rationalize SHOW TRACE

Column label renamed as follows:

- `loc` -> `location` (avoidance of abbreviation)

Release note (sql change): The output label of `SHOW TRACE` has been
renamed for consistency with other SHOW statements.

knz committed:
sql: rationalize SHOW CONSTRAINTS

The column labels are renamed as follows:

- `table` -> `table_name` (casing, consistency, keyword avoidance, disambiguation)
- `name` -> `constraint_name` (casing, disambiguation)
- `type` -> `constraint_type` (casing, disambiguation)

Release note (sql change): The output labels of `SHOW CONSTRAINTS` are
renamed for consistency with other SHOW statements.

knz committed:
sql: rationalize SHOW SESSION ALL

Prior to this patch, SHOW SESSION ALL would be a pass-through to
`crdb_internal.session_variables`. This is susceptible to API breakage
when the (internal, unstable) vtable is updated.

This patch removes this risk by fixing the columns produced by SHOW
SESSION ALL.

Release note: None

knz committed:
sql: rationalize SHOW ALL CLUSTER SETTINGS

The output labels are renamed as follows:

- `name` -> `variable` (consistency with SHOW SESSION ALL)
- `current_value` -> `value` (consistency with SHOW SESSION ALL)
- `type` -> `setting_type` (disambiguation)

Release note (sql change): The output labels of `SHOW ALL CLUSTER
SETTINGS` have been renamed for consistency with the other SHOW
statements.

knz committed:
sql: rationalize SHOW EXPERIMENTAL_FINGERPRINTS

The column label "index" is renamed to "index_name" to avoid producing
a keyword-named column, and for consistency with other SHOW statements.

Release note: None

knz committed:
sqlccl: rationalize SHOW BACKUP

The output labels are renamed as follows:

- `database` -> `database_name` (disambiguation)
- `table` -> `table_name` (keyword avoidance, disambiguation)

Release note (sql change): The output labels of `SHOW BACKUP` are
renamed for consistency with the other SHOW statements.

knz committed:
sql: rationalize SHOW JOBS

The output labels are renamed as follows:

- `id` -> `job_id` (disambiguation)
- `type` -> `job_type` (disambiguation)
- `username` -> `user_name` (space between words)

This patch also renames the columns in `crdb_internal.jobs`
accordingly.

Release note (sql change): The output labels of `SHOW JOBS` are
renamed for consistency with other SHOW statements.

RaduBerinde commented:
Thanks for taking this on! LGTM!

Regarding the new simplified `SHOW CREATE` syntax - is it possible for the object type to be ambiguous? If yes, it would be good to add a logictest.

knz committed:
sql: rationalize SHOW EXPERIMENTAL ZONE CONFIGURATION

The column labels are renamed as follows:

- `id` -> `zone_id` (disambiguation)
- `config_proto` -> `config_protobuf` (abbreviation avoidance)

This patch also renames the columns in `crdb_internal.zones`
accordingly.

Release note: None

knz committed:
sql: rationalize SHOW HISTOGRAM

The column labals are renamed as follows:

- `num_range` -> `range_rows` (abbreviation avoidance, consistency
  with equivalent feature in other RDBMS)
- `num_eq` -> `equal_rows` (abbreviation avoidance, consistency
  with equivalent feature in other RDBMS)

Release note (sql change): The output labels for `SHOW HISTOGRAM` are
renamed for consistency with other SHOW statements.

knz committed:
sql: rationalize SHOW STATISTICS

The column labels are renamed as follows:

- `name` -> `table_name` (disambiguation)
- `columns` -> `column_names` (disambiguation)
- `created_at` -> `created` (consistency with SHOW JOBS)

Release note (sql change): The output labels of `SHOW STATISTICS` are
renamed for consistency with other SHOW statements.

knz committed:
sql: rationalize SHOW USERS

Release note (sql change): The output label of `SHOW USERS` is renamed
for consistency with the other SHOW statements.

Release note (cli change): The output labels of `cockroach user ls`
and `cockroach user get` are renamed for consistency with the SQL
`SHOW USERS` statement. The output of `cockroach user get` does not
include the hashed password any more, to reduce inadvertent data
leaks.

knz committed:
docs: explain SQL column naming guidelines in the style guide

Release note: None

knz commented:
@RaduBerinde can you clarify what you mean with 

> is it possible for the object type to be ambiguous? 

The syntax is either `SHOW CREATE X Y` or `SHOW CREATE Y`. The number of tokens is different so `X` is always a keyword and `Y` always a name.

RaduBerinde commented:
I mean if there are two objects of different types that have the same name.

knz commented:
> I mean if there are two objects of different types that have the same name.

Nope that's not possible (neither in pg nor in cockroachdb) - tables, sequences and views share the same namespace.

knz commented:
:tada: with my recent SQLAlchemy updates, the examples-orms tests are now passing. I'll rebase this and merge soon unless anyone objects.

knz commented:
bors r+

knz commented:
bors r-

(forgot an edit)

knz commented:
bors r+

jseldess commented:
Docs updated in https://github.com/cockroachdb/docs/pull/3523, with the exception of `EXPLAIN`, which will be handled separately.
